Improvements to integration URL modal.

sbansal1999 opened the pull request:
This PR makes the following improvements to the integration URL modal:

- Migrates the integration input from typeahead to a dropdown widget.
- Disables the `Copy URL` button when the topic input is empty

Fixes: #26797

<hr>


**Screenshots and screen captures:**

<details> 


<summary>Integration input </summary>


| Before | After |
|--------|--------|
| ![typeahead](https://github.com/sbansal1999/zulip/assets/35286603/28655e9d-51b6-4965-a8eb-7740a9062654) | ![dropdown widget](https://github.com/sbansal1999/zulip/assets/35286603/154de25b-8bdc-48d1-8dc2-32892ef6e2b8) | 


</details>

<details>

<summary> "Copy URL" button when topic input is empty </summary>

| Before | After |
|--------|--------|
| ![image](https://github.com/sbansal1999/zulip/assets/35286603/db202417-0401-4784-b111-6ff51a046224)  | ![image](https://github.com/sbansal1999/zulip/assets/35286603/b56e1f02-5d8d-4432-a895-28a66df166ba)  | 

</details>

<details>

<summary> Mobile view </summary>

![image](https://github.com/zulip/zulip/assets/35286603/05958a9f-aeec-46e9-b582-b97356b8f2ce)

</details>

<hr>

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on zerver/openapi/zulip.yaml:
```diff
@@ -11684,8 +11684,8 @@ paths:
-                          An array of dictionaries where each dictionary describes a type of incoming webhook
-                          integration that is available to be configured on this Zulip server.
+                          An array of dictionaries where each dictionary describes a type of non-legacy incoming
+                          webhook integration that is available to be configured on this Zulip server.
```
This probably doesn't require an API feature level update or API documentation change -- we're just changing the meaning of "available to be configured on this server"?

This commit's comment message include an explanation for why we're excluding these, and possibly also that should go in a comment in `zerver/lib/events.py`.

timabbott commented:
Posted one comment, but otherwise this lgtm (not having tested at all).

sbansal1999 commented:
**Updates:**

- Removed the API version as well as the documentation change.
- Added a comment mentioning why the change was required as suggested in [this](https://github.com/zulip/zulip/pull/26956#discussion_r1342151873) comment.

alya commented:
Looks great to me in manual testing!

timabbott commented:
Merged after a small correction in the `events.py` comment about the meaning of legacy integrations. Thanks @sbansal1999!
